packages/circuit-ui/components/Dialog/Dialog.stories.tsx

```diff
@@ -0,0 +1,204 @@
+    name: 'Open modal',
+  });
+
+  await userEvent.click(button);
+  await screen.findByRole('dialog');
+};
+
+const baseArgs: DialogProps = {
+  open: false,
+  onClose: () => {},
+  'aria-labelledby': 'title',
+  'aria-describedby': 'description',
+  variant: 'contextual',
+  closeButtonLabel: 'Close',
+  children: defaultModalChildren,
+};
+
+export const Base = (modal: DialogProps) => {
+  const [modalOpen, setModalOpen] = useState(false);
+  return (
+    <>
+      <Button
+        type="button"
+        onClick={() => {
+          setModalOpen(true);
+        }}
+      >
+        Open modal
+      </Button>
+      <Dialog {...modal} open={modalOpen} onClose={() => setModalOpen(false)} />
+    </>
+  );
+};
+Base.args = baseArgs;
+Base.play = openModal;
+
+export const WithUseModal = (modal: DialogProps) => {
+  const ComponentWithModal = () => {
+    const { setModal } = useModal();
+
+    return (
+      <Button type="button" onClick={() => setModal(modal)}>
+        Open modal
+      </Button>
+    );
+  };
+  return (
+    <ModalProvider>
+      <ComponentWithModal />
+    </ModalProvider>
+  );
+};
+WithUseModal.args = {
+  children: defaultModalChildren,
+  closeButtonLabel: 'Close modal',
+};
+WithUseModal.play = openModal;
+
+export const InitiallyOpen = (modal: DialogProps) => {
+  const initialModal = { id: 'initial', component: Dialog, ...modal };
+
+  return (
+    <ModalProvider initialState={[initialModal]}>
+      <div />
+    </ModalProvider>
+  );
+};
+InitiallyOpen.args = {
+  children: defaultModalChildren,
+  variant: 'contextual',
+  closeButtonLabel: 'Close modal',
+};
+
+export const Immersive = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  return (
+    <>
+      <Button
+        type="button"
+        onClick={() => {
+          setModalOpen(true);
+        }}
+      >
+        Open modal
+      </Button>
+      <Dialog
+        open={modalOpen}
+        onClose={() => setModalOpen(false)}
+        closeButtonLabel="Close"
+        aria-labelledby="title"
+        aria-describedby="description"
+        variant="immersive"
+      >
+        {defaultModalChildren}
+      </Dialog>
+    </>
+  );
+};
+Immersive.parameters = {
+  chromatic: { disableSnapshot: true },
+  viewport: {
+    defaultViewport: 'smallMobile',
+  },
+};
+Immersive.play = openModal;
+
+export const PreventClose = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  return (
+    <>
+      <Button
+        type="button"
+        onClick={() => {
+          setModalOpen(true);
+        }}
+      >
+        Open modal
+      </Button>
+      <Dialog
+        open={modalOpen}
+        preventClose
+        onClose={() => setModalOpen(false)}
+        closeButtonLabel="Close"
+        aria-labelledby="title"
+        aria-describedby="description"
+      >
+        {defaultModalChildren}
+      </Dialog>
+    </>
+  );
+};
+PreventClose.parameters = {
+  chromatic: { disableSnapshot: true },
```

Comment:
I appreciate that you customized the Chromatic configuration for individual stories 👏🏻